### ISSUE TITLE
Global vs local scope - better explanation - shadowing instead of overiding

### ISSUE THREAD

**Issue by msanecki** [FreeCodeCamp/FreeCodeCamp#11396]
It's about the Instruction. There is a term "override" used to explain the case.

I'd suggest to change it to "shadow". Please look at this explanation:
http://stackoverflow.com/questions/5373278/variable-shadowing-in-javascript

**Comment by dhcodes** [comment]
While true, I'd worry that this adds confusion for how the global vs. local scope appears to work and the term "shadow" obfuscates the idea imho.

**Comment by erictleung** [comment]
I really like the analogy of "shadowing" variables and I feel that it more aptly describes what is going in local vs global scope.

However, I agree with @dhcodes that is might confuse new coders. Because if we mention "shadowing," we would have to take the time to describe what that means.

The word "override" is a bit more common in everyday language and immediately understandable to our target audience, which is generally new to all programming concepts.

**Comment by msanecki** [comment]
I think it important especially for greenhorns to get the base concepts right from the beginning. 
@erictleung it's not about what is common but what is right, right? ;) Shadowing is also used in everyday language, the more sun the more use of the word 
The global var stay in the shadow of local one because of the function scope....

**Comment by BerkeleyTrue** [comment]
> I think it important especially for greenhorns to get the base concepts right from the beginning. 

This is a common misconception and it usually hinders learning by introducing too many concepts to soon or to quickly.

It important that every challenge is teaching only one concept.

> Shadowing is also used in everyday language,

Which language? My concept of shadowing (in everyday use) is probably different from yours and is not applicable to this situation. Also, remember that the concept of shadowing is going to be new regardless since we are trying to teach it in the context of programing. We must take small steps first before we can leap forward.

override is a common english word and it's definition is easily applicable in this situation.

**Comment by msanecki** [comment]
> This is a common misconception and it usually hinders learning by introducing too many concepts to soon or to quickly.

Please, to quick or to soon? look at the challenges in every second there is some new concept or construct.

> It important that every challenge is teaching only one concept.
In my opinion this challenge teach exactly one concept, the concept of shadowing.

@BerkeleyTrue what are you going to say if someone ask you about the difference between shadowing and overriding based on this stuff?

**Comment by erictleung** [comment]
@msanecki using your argument on "it's not about what is common but what is right," the Stack Overflow post you initially linked to quotes the following.

> In computer programming, variable shadowing occurs when a variable declared within a **certain scope** (decision block, method, or inner class) has the same name as a variable declared in a **outer scope**.

The concept really being taught here is to extend your understanding of the concept of different scopes and how they play together. The previous two lessons introduce them individually, and this lesson hopes to use them together without introducing too many new things. Nonetheless, I think the 

Once again, the target audience for this point in the curriculum is individuals who have little to no programming/computer experience. We're all biased for programming for at least a little bit to be comfortable. So my vision of the curriculum is to have the lowest entry to web development as possible. Nonetheless, I think the challenge still teaches the concept we're all talking about through code. Just my two cents.

We can leave this open for others (experienced to novice programmers) to chime in. Otherwise, this might be bikeshedding a bit.

Possible other solution: write a Wiki entry for "variable shadowing" or edit the current global and local scope Wiki to include "variable shadowing"

**tl;dr** I would vote for being less pedantic and more pragmatic and just focus on the concept of scope.

**Comment by msanecki** [comment]
> So my vision of the curriculum is to have the lowest entry to web development as possible. 

@erictleung I like your point of view. From my point of view the issue can get closed. I could write a wiki entry explaining this. Where can I find the wiki?

**Comment by dhcodes** [comment]
@msanecki [Here's the link](http://forum.freecodecamp.com/c/wiki) to the wiki, which is currently housed on the forum. You may need a few forum posts in order to be able to gain edit rights to the wiki. I'm not sure.

Specific wiki page for that challenge: http://forum.freecodecamp.com/t/challenge-global-vs-local-scope-in-functions/18194

Thanks all who contributed to the discussion.
